code/modules/spells/targeted/necromancer_spells.dm

```diff
@@ -141,15 +141,62 @@
 	hud_state = "wiz_lich"
 
 /spell/targeted/lichdom/cast(mob/target,var/mob/living/carbon/human/user as mob)
+
 	if(isskeleton(user))
 		user << "You have no soul or life to offer."
 		return
 
 	user.visible_message("<span class='cult'>\The [user]'s skin sloughs off bone, their blood boils and guts turn to dust!</span>")
 	gibs(user.loc)
-	user.verbs += /mob/living/carbon/proc/immortality
+	user.verbs += /mob/living/carbon/proc/dark_resurrection
 	user.set_species("Skeleton")
 	user.unEquip(user.wear_suit)
 	user.unEquip(user.head)
 	user.equip_to_slot_or_del(new /obj/item/clothing/suit/wizrobe/black(user), slot_wear_suit)
 	user.equip_to_slot_or_del(new /obj/item/clothing/head/wizard/black(user), slot_head)
+	user << "<span class='notice'>Your soul flee to the remains of your heart, turning it into your phylactery. Do not allow anyone to destroy it!</span>"
+	var/obj/item/phylactery/G = new(get_turf(user))
+	G.lich = user
+	G.icon_state = "cursedheart-on"
+
+/mob/living/carbon/proc/dark_resurrection()
+	set category = "Abilities"
+	set name = "Dark Resurrection"
+	set desc = "Return to your phylactery and rebuild your body."
+
+	var/mob/living/carbon/C = usr
+	if(!C.stat)
+		C << "<span class='notice'>You're not dead yet!</span>"
+		return
+		
+	var/obj/item/phylactery/P
+	for(var/thing in world_phylactery)
+		var/obj/item/phylactery/N = thing
+		if (!QDELETED(N) && N.lich == C)
+			P = N
+		 
+	if(P)
+		C.forceMove(get_turf(P))
+		C << "<span class='notice'>Your dead body returns to your phylactery, slowly rebuilding itself.</span>"
+		C.verbs -= /mob/living/carbon/proc/dark_resurrection
+		addtimer(CALLBACK(src, .proc/post_dark_resurrection), rand(400, 800))
+
+	else
+		C << "<span class='danger'>Your phylactery was destroyed, your existence will face oblivion now.</span>"
+		C.visible_message("<span class='cult'>\The [C]'s body turns into dust, a twisted wail can be heard!</span>")
```

Comment:
The grammar here is a little dicey. Perhaps `As [C]'s body turns to dust, a twisted wail can be heard!`